urls.go

```diff
@@ -30,6 +30,9 @@ import (
 	"strings"
 )
 
+// Max length of the scpUrl to prevent reDOS attacks
```

Comment:
You should link to https://nvd.nist.gov/vuln/detail/CVE-2023-46402 and https://gist.github.com/6en6ar/7c2424c93e7fbf2b6fc44e7fb9acb95d in this comment so people reading the code in the future have additional context.